pkgs/applications/editors/vscode/common.nix

```diff
@@ -0,0 +1,78 @@
+# Define some common items among all vscode variants.
+{
+  lib,
+  makeDesktopItem,
+  executableName,
+  longName,
+  shortName,
+}:
+{
+  desktopItem = makeDesktopItem {
+    name = executableName;
+    desktopName = longName;
+    comment = "Code Editing. Redefined.";
+    genericName = "Text Editor";
+    exec = "${executableName} %F";
+    icon = "vs${executableName}";
+    startupNotify = true;
+    startupWMClass = shortName;
+    categories = [
+      "Utility"
+      "TextEditor"
+      "Development"
+      "IDE"
+    ];
+    mimeTypes = [
+      "text/plain"
+      "inode/directory"
+    ];
+    keywords = [ "vscode" ];
+    actions.new-empty-window = {
+      name = "New Empty Window";
+      exec = "${executableName} --new-window %F";
+      icon = "vs${executableName}";
+    };
+  };
+
+  urlHandlerDesktopItem = makeDesktopItem {
+    name = executableName + "-url-handler";
+    desktopName = longName + " - URL Handler";
+    comment = "Code Editing. Redefined.";
+    genericName = "Text Editor";
+    exec = executableName + " --open-url %U";
+    icon = "vs${executableName}";
+    startupNotify = true;
+    categories = [
+      "Utility"
+      "TextEditor"
+      "Development"
+      "IDE"
+    ];
+    mimeTypes = [ "x-scheme-handler/vs${executableName}" ];
+    keywords = [ "vscode" ];
+    noDisplay = true;
+  };
+
+  meta = {
+    description = ''
+      Open source source code editor developed by Microsoft for Windows,
+      Linux and macOS
+    '';
+    longDescription = ''
+      Open source source code editor developed by Microsoft for Windows,
```

Comment:
longDescription should not repeat description, as they are often displayed very close to each other.

Reply:
Same as above, these words have already presented in nixpkgs before:

https://github.com/NixOS/nixpkgs/blob/1032df05d070c7102304f71930b39c505ea226d6/pkgs/applications/editors/vscode/vscode.nix#L91